tests/CMakeLists.txt

```diff
@@ -44,6 +26,36 @@ target_link_libraries(avifyuv avif ${AVIF_PLATFORM_LIBRARIES})
 
 add_custom_target(avif_test_all COMMAND $<TARGET_FILE:aviftest> ${CMAKE_CURRENT_SOURCE_DIR}/data DEPENDS aviftest)
 
+if(AVIF_BUILD_GTESTS)
+    enable_language(CXX)
+    set(CMAKE_CXX_STANDARD 11)
+    set(GTEST_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/ext/googletest/googletest/include)
+    set(GTEST_LIBRARIES ${CMAKE_SOURCE_DIR}/ext/googletest/build/lib/libgtest${CMAKE_STATIC_LIBRARY_SUFFIX})
+    set(GTEST_MAIN_LIBRARIES ${CMAKE_SOURCE_DIR}/ext/googletest/build/lib/libgtest_main${CMAKE_STATIC_LIBRARY_SUFFIX})
+    set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
+    if(EXISTS ${GTEST_INCLUDE_DIRS}/gtest/gtest.h AND EXISTS ${GTEST_LIBRARIES} AND EXISTS ${GTEST_MAIN_LIBRARIES})
+        message(STATUS "Found local ext/googletest")
+    else()
+        message(STATUS "Missing local ext/googletest, falling back to finding installed GoogleTest")
+        find_package(GTest REQUIRED)
+    endif()
+
+    if(CMAKE_C_COMPILER_ID MATCHES "Clang")
+        # -Weverything is mostly fine for C but triggers too many C++ warnings in GoogleTest.
+        add_compile_options(-Wno-everything) # Applied to current directory only.
+    endif()
+
```

Comment:
The help message for the `AVIF_BUILD_GTESTS` option says "Requires GoogleTest to be installed and AVIF_BUILD_TESTS to be ON." Should we enforce these two requirements here?

Reply:
I added a message when `AVIF_BUILD_GTESTS` is `ON` but `AVIF_ENABLE_GTEST` is `OFF`.

I added a warning when `AVIF_BUILD_GTESTS` is `OFF` but `AVIF_ENABLE_GTEST` is `ON`.